pyclam/manifold.py

```diff
@@ -608,6 +607,31 @@ def __repr__(self) -> str:
     def depth(self) -> int:
         return len(self.graphs) - 1
 
+    def distance(self, x1: Union[List[int], Data], x2: Union[List[int], Data]) -> np.ndarray:
+        """ Calculates the pairwise distances between all points in x1 and x2.
+
+        This DOES NOT do any batching.
+
+        The metric given to Manifold should have the following properties:
+            * dist(p1, p2) = 0 if and only if p1 = p2.
+            * dist(p1, p2) = dist(p2, p1)
+
+        :param x1: a list of indices, or a 2D matrix of data points
+        :param x2: a list of indices, or a 2D matrix of data points
+        :return: matrix of pairwise distances.
+        """
+
+        x1, x2 = np.asarray(x1), np.asarray(x2)
+        
+        # Fetch data if given indices.
+        if len(x1.shape) < 2:
+            x1 = self.data[x1 if x1.ndim == 1 else np.expand_dims(x1, 0)]
```

Comment:
This is wrong.
When x1.shape is less than 2, x1.ndim is always 1

Reply:
No, it can be zero. `len(np.array(1))`